examples/adjoints/adjoints_ex5/heatsystem.h

```diff
@@ -55,7 +55,16 @@ class HeatSystem : public FEMSystem
   // Sensitivity Calculation
   Number & compute_final_sensitivity()
   {
-    final_sensitivity = -(R_plus_dp - R_minus_dp)/(2*dp);
+    final_sensitivity = 0.0;
+
+    // Use the trapezoidal rule to compute the sensitivity integral
```

Comment:
Good catch with this, but it's only correct here because we're doing first-order time stepping with constant deltat.  In the future we ought to try to figure out some way to abstract this into the TimeSolver (for the integration rule) combined with the SolutionHistory (for the deltat changes).

Reply:
IMO, we should further abstract our adjoint senstivity methods to a class like we do for error estimation, where the right function to compute sensitivity is used depending on the TimeSolver and SolutionHistory.